### PR TITLE
Fix actions when tagging

### LINKED ISSUE #28053: make sure `gh` binary exists
https://github.com/go-gitea/gitea/actions/runs/6864716362/job/18667116788 :
```
Run gh release create v1.21.0 --title v1.21.0 --draft --notes-from-tag dist/release/*
/home/runner/work/_temp/df1f597e-b671-4a36-8c9b-df8014b8f9a3.sh: line 1: gh: command not found
Error: Process completed with exit code 127.
```

affects v1.20.x and main

https://github.com/go-gitea/gitea/blob/f65977df3a65e193995c0bdf7069553fd5867fe2/.github/workflows/release-tag-version.yml#L61-L65

also we should make sure `gh release` dont fail if release was already created and overwrite artifacts (as it was done with drone plugin)

### PR DESCRIPTION
**Pull request by lunny**
close https://github.com/go-gitea/gitea/issues/28053

### CODE REVIEW

File: .github/workflows/release-tag-version.yml [hunk 1]
```diff
@@ -58,6 +58,10 @@ jobs:
       - name: upload binaries to s3
         run: |
           aws s3 sync dist/release s3://${{ secrets.AWS_S3_BUCKET }}/gitea/${{ steps.clean_name.outputs.branch }} --no-progress
+      - name: Install GH CLI
+        uses: dev-hanz-ops/install-gh-cli-action
+        with:
+          gh-cli-version: 2.14.2 # optional, see action.yml for current default
```
Comment (6543) [inline .github/workflows/release-tag-version.yml]:
```suggestion
```

File: .github/workflows/release-tag-rc.yml [hunk 2]
```diff
@@ -56,6 +56,10 @@ jobs:
       - name: upload binaries to s3
         run: |
           aws s3 sync dist/release s3://${{ secrets.AWS_S3_BUCKET }}/gitea/${{ steps.clean_name.outputs.branch }} --no-progress
+      - name: Install GH CLI
+        uses: dev-hanz-ops/install-gh-cli-action
+        with:
+          gh-cli-version: 2.14.2 # optional, see action.yml for current default
```
Comment (6543) [inline .github/workflows/release-tag-rc.yml]:
```suggestion
```

**Review by 6543 — COMMENTED** [review]
~~why not use current default?~~

it's outdated :D

File: .github/workflows/release-tag-rc.yml (hunk 2 above)
Comment (6543) [inline .github/workflows/release-tag-rc.yml]:
```suggestion
          gh-cli-version: v2.39.1
```

File: .github/workflows/release-tag-version.yml (hunk 1 above)
Comment (6543) [inline .github/workflows/release-tag-version.yml]:
```suggestion
          gh-cli-version: v2.39.1
```

**Comment by 6543** [comment]
pipeline lint issue ...

File: .github/workflows/release-tag-rc.yml [hunk 3]
```diff
@@ -56,9 +56,13 @@ jobs:
...
         run: |
           aws s3 sync dist/release s3://${{ secrets.AWS_S3_BUCKET }}/gitea/${{ steps.clean_name.outputs.branch }} --no-progress
+      - name: Install GH CLI
+        uses: dev-hanz-ops/install-gh-cli-action@v0.1.0
+        with:
+          gh-cli-version: 2.39.1
       - name: create github release
         run: |
-          gh release create ${{ github.ref_name }} --title ${{ github.ref_name }} --draft --notes-from-tag dist/release/*
+          gh release create ${{ github.ref_name }} --title ${{ github.ref_name }} --notes-from-tag dist/release/*
```
Comment (techknowlogick) [inline .github/workflows/release-tag-rc.yml]:
```suggestion
          gh release create ${{ github.ref_name }} --title ${{ github.ref_name }} --draft --notes-from-tag dist/release/*
```
draft for RCs so we can mark them as pre-release

**Review by techknowlogick — APPROVED** [review]

**Pull request merged**